### PR TITLE
Added kotlin support to :app module

### PR DESCRIPTION
**Pull request by gordwilling**
Added Kotlin support to :app module
Updated some android API dependencies to recommended latest versions
Migrated Export.java to Export.kt

This change involved generating a String using Kotlin instead of Java. There was a unit test in place that successfully ran to verify correctness. I did not see the need to deploy to an emulator.

This is my first ever pull request on GitHub :)

### CODE REVIEW

**Comment by VREMSoftwareDevelopment** [comment]
Thanks for pull request.

- Please make sure that you are using feature branch, since all the WiFi Analyzer changes are done on the feature branch!
- WiFiAnalyzer master branch is only used to release new versions of the application. As a result, the master branch most likely doesn't contain the latest changes.

[How to submit a pull request](https://github.com/VREMSoftwareDevelopment/WiFiAnalyzer/wiki/Pull-Request)

File: app/src/main/kotlin/com/vrem/wifianalyzer/export/Export.kt [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+val header = String.format(ENGLISH,
+        "Time Stamp|SSID|BSSID|Strength|Primary Channel|Primary Frequency|Center Channel|Center Frequency|Width (Range)|Distance|802.11mc|Security%n")
+
+fun getData(wiFiDetails: List<WiFiDetail>, timestamp: String) =
+        header + wiFiDetails.joinToString(
+                System.lineSeparator(),
+                transform = toExportString(timestamp))
+
+fun toExportString(timestamp: String) = { wiFiDetail: WiFiDetail ->
+    with(wiFiDetail) {
```
Comment (VREMSoftwareDevelopment) [inline app/src/main/kotlin/com/vrem/wifianalyzer/export/Export.kt]: wiFiDetail is never null

File: app/src/test/java/com/vrem/wifianalyzer/export/ExportTest.java [hunk 2]
```diff
@@ -33,26 +32,19 @@
 
 public class ExportTest {
```
Comment (VREMSoftwareDevelopment) [inline app/src/test/java/com/vrem/wifianalyzer/export/ExportTest.java]: Any plans to convert test from Java to KT?

Comment (VREMSoftwareDevelopment) [inline app/src/test/java/com/vrem/wifianalyzer/export/ExportTest.java]: Any plans to convert test from Java to KT?

File: app/src/main/kotlin/com/vrem/wifianalyzer/export/Export.kt (hunk 1 above)
Comment (gordwilling) [reply app/src/main/kotlin/com/vrem/wifianalyzer/export/Export.kt]: That's the _with_ function. Kotlin In Action, p. 128 ;)

**Pull request merged**